### PR TITLE
cherry-pick commit

### PR DESCRIPTION
**Pull request by PakhiAgarwal**

### CODE REVIEW

**Comment by PakhiAgarwal** [comment]
Please have a look and let me know if it is good. There are duplicate files in test, which I'll remove if it is good.

**Comment by merlinvn** [comment]
It is the same as other pull requests where you still have files that are not inside your 'test' folder appear in the pull request file changes. Although those files just contain the format changes but it is not accepted, because in future, I do not have time to scan and verify all the unnecessary format changes. Those files are:  'CMakeLists.txt', 'src/CMakeLists.txt '. And also you deleted 1 file in the 'test/old' folder.

In other words, you cannot make a pull request that contains your dev settings / changes as it will changes/delete files outside of your scope in the main repo.

**Comment by PakhiAgarwal** [comment]
There are so many files with changes  because it is the first time I am trying to commit my changes to this repo. In the future I'll send the pull request as soon as it is done for every file.

As there are only format changes not, it would be great if you could accept either this or previous pull request. In the future I'll make sure there are no conflicts and I make changes only in test folder.

Because I have so many commits, so the only thing we can do is either you can accept the pull request or I can delete this repo as it is having all my commits and all my branches are branched with the settings of `spatial` branch which I was asked to work on. Deleting this repo will create problems with my settings of test code because of which I don't want to delete. 

The previous pull request is not having any conflicts and I think it could be merged. It is also not interfering with any of your settings. 

One other thing you can do is to create a branch in this repo with your settings and grant me access for the branch to merge commits on my own. I will only merge my test folder in that branch.

**Comment by merlinvn** [comment]
You don't understand the concept of having a 'clean' pull request. If you
delete the whole 'src' folder and make a pull request, you will have a pull
request without any conflict, but that PR is not 'clean'.

If a clean pull request has conflicts, then we can resolve those, but it's
another story.

I told you couple times before is to create a dedicate branch from 'dev'
just for pull request, please think about it.
